### PR TITLE
Fix compilation in Geometry/MTDCommonData for gcc10 and 9

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:

Geometry/MTDCommonData fails to compile in gcc10 and gcc9 
because of 
```
  /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/915ba8a2f00f0d1d62c35486d637e2be/opt/cmssw/slc7_amd64_gcc10/cms/cmssw/CMSSW_11_2_X_2020-09-24-2300/src/Geometry/MTDCommonData/src/ETLNumberingScheme.cc:85:98: error: comparison of integer expressions of different signedness: 'const int' and 'const uint32_t' {aka 'const unsigned int'} [-Werror=sign-compare]
    85 |       (!preTDR && (1 > modCopy || std::max(ETLDetId::kETLv4maxModule, ETLDetId::kETLv5maxModule) < modCopy))) {
```
see 
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc10/CMSSW_11_2_X_2020-09-24-2300/Geometry/MTDCommonData
std::max returns int and modCopy is uint_32. 
It looks like this PR did it
https://github.com/cms-sw/cmssw/pull/31511

#### PR validation:

builds with gcc10 IB

### CODE REVIEW

**Comment by mrodozov** [comment]
test parameters:
- architecture=slc7_amd64_gcc10

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
when I checked out the entire Geometry and rebuild it (gcc10 IB) the GeometryMTDCommonDataTestDriver test ran.
I can rebuild with "full cmssw -> ON" to make that point but I don't think it's necessary.

**Comment by kpedro88** [comment]
+upgrade

File: Geometry/MTDCommonData/src/ETLNumberingScheme.cc [hunk 1]
```diff
@@ -82,7 +82,8 @@ uint32_t ETLNumberingScheme::getUnitID(const MTDBaseNumber& baseNumber) const {
   }
 
   if ((preTDR && (1 > modCopy || ETLDetId::kETLv1maxModule < modCopy)) ||
-      (!preTDR && (1 > modCopy || std::max(ETLDetId::kETLv4maxModule, ETLDetId::kETLv5maxModule) < modCopy))) {
+      (!preTDR &&
+       (1 > modCopy || (unsigned)std::max(ETLDetId::kETLv4maxModule, ETLDetId::kETLv5maxModule) < modCopy))) {
```
Comment (cvuosalo) [inline Geometry/MTDCommonData/src/ETLNumberingScheme.cc:86]:
```suggestion
       (1 > modCopy || static_cast<unsigned>(std::max(ETLDetId::kETLv4maxModule, ETLDetId::kETLv5maxModule)) < modCopy))) {
```

Comment (cvuosalo) [reply Geometry/MTDCommonData/src/ETLNumberingScheme.cc:86]: `static_cast` is preferred.

**Comment by smuzaffar** [comment]
please test

**Comment by cvuosalo** [comment]
+1

**Comment by smuzaffar** [comment]
@silviodonato , @qliphy can we please get this in? this should fix the gcc 9 and gcc10 IBs build errors

**Comment by smuzaffar** [comment]
@kpedro88 , can you please re-sign this?

**Comment by silviodonato** [comment]
merge
let's get a working IB for gcc10 and gcc9. If there are further comments, we can implement them later

**Comment by kpedro88** [comment]
+upgrade

**Pull request merged**